### ISSUE TITLE
Run `build-linux` and `build-osx` in parallel?

### ISSUE THREAD

**Issue by niemasd** [bioconda/bioconda-recipes#46294]
Right now, I'm in the process of trying to debug a PR I'm creating:

https://github.com/bioconda/bioconda-recipes/pull/46292

I have it successfully building for Linux, but I need to patch it to build for OS X. However, what's a bit frustrating is that, because `build-osx` job has `needs: build-linux`, I have to wait ~40 minutes for the entire `build-linux` job to succeed before getting to the `build-osx` job:

https://github.com/bioconda/bioconda-recipes/blob/ccdeb5760ceb59e1db08a52f61cdfb225b518306/.github/workflows/PR.yml#L172C5-L172C23

Would it be possible to change the `build-osx` job from `needs: build-linux` to `needs: lint` (i.e., the same dependency as `build-linux`) so that `build-osx` and `build-linux` could run in parallel?

https://github.com/bioconda/bioconda-recipes/blob/ccdeb5760ceb59e1db08a52f61cdfb225b518306/.github/workflows/PR.yml#L68

**Comment by martin-g** [comment]
I think you could try the proposed change in your PR and if everything works as desired then you can propose it in its own PR!
